index.js

```diff
@@ -82,6 +101,14 @@ const preprocessDestinationPath = ({entry, destination, options}) => {
 		return path.join(destination, entry.name);
 	}
 
+	if (entry.pattern.isDirectory && path.relative(options.cwd, entry.path).startsWith('..')) {
+		return path.join(options.cwd, destination, path.basename(entry.pattern.originalPath), path.relative(entry.pattern.originalPath, entry.path));
+	}
+
+	if (!entry.pattern.isDirectory && entry.path === entry.relativePath) {
```

Comment:
It would be good to have a short code comment on both of these `if`'s on what kind of case they handle.